astropy/io/ascii/tests/test_read.py

```diff
@@ -1284,3 +1286,82 @@ def test_unsupported_read_with_encoding(tmpdir):
     with pytest.raises(ascii.ParameterError):
         ascii.read('t/simple3.txt', guess=False, fast_reader='force',
                    encoding='latin1', format='fast_csv')
+
+
+def test_read_chunks_input_types():
+    """
+    Test chunked reading for different input types: file path, file object,
+    and string input.
+    """
+    fpath = 't/test5.dat'
+    t1 = ascii.read(fpath, header_start=1, data_start=3, )
+
+    for fp in (fpath, open(fpath, 'r'), open(fpath, 'r').read()):
+        t_gen = ascii.read(fp, header_start=1, data_start=3,
+                           guess=False, format='fast_basic',
+                           fast_reader={'chunk_size': 400, 'chunk_generator': True})
+        ts = list(t_gen)
+        for t in ts:
+            for col, col1 in zip(t.columns.values(), t1.columns.values()):
+                assert col.name == col1.name
+                assert col.dtype.kind == col1.dtype.kind
+
+        assert len(ts) == 4
+        t2 = table.vstack(ts)
+        assert np.all(t1 == t2)
+
+    for fp in (fpath, open(fpath, 'r'), open(fpath, 'r').read()):
+        # Now read the full table in chunks
+        t3 = ascii.read(fp, header_start=1, data_start=3,
+                        fast_reader={'chunk_size': 300})
+        assert np.all(t1 == t3)
+
+
+@pytest.mark.parametrize('masked', [True, False])
+def test_read_chunks_formats(masked):
+    """
+    Test different supported formats for chunked reading.
+    """
+    t1 = simple_table(size=102, cols=10, kinds='fS', masked=masked)
+    for i, name in enumerate(t1.colnames):
+        t1.rename_column(name, 'col{}'.format(i + 1))
+
+    # TO DO commented_header does not currently work due to the special-cased
+    # implementation of header parsing.
+
+    for format in 'tab', 'csv', 'no_header', 'rdb', 'basic':
+        out = StringIO()
+        ascii.write(t1, out, format=format)
+        t_gen = ascii.read(out.getvalue(), format=format,
+                           fast_reader={'chunk_size': 400, 'chunk_generator': True})
+        ts = list(t_gen)
+        for t in ts:
+            for col, col1 in zip(t.columns.values(), t1.columns.values()):
+                assert col.name == col1.name
+                assert col.dtype.kind == col1.dtype.kind
+
+        assert len(ts) > 4
+        t2 = table.vstack(ts)
+        assert np.all(t1 == t2)
+
+        # Now read the full table in chunks
+        t3 = ascii.read(out.getvalue(), format=format, fast_reader={'chunk_size': 400})
+        assert np.all(t1 == t3)
+
+
+def test_read_chunks_chunk_size_too_small():
+    fpath = 't/test5.dat'
+    with pytest.raises(ValueError) as err:
+        ascii.read(fpath, header_start=1, data_start=3,
+                   fast_reader={'chunk_size': 10})
+    assert 'no newline found in chunk (chunk_size too small?)' in str(err)
+
+
+@pytest.mark.xfail
```

Comment:
How about catching the exception and checking it instead of marking xfail? To me, xfail means "it's a test that we currently expect to fail, but we will fix that". In the output of running the tests, it's marked with "x" which is not the same a passed test ("-").

Reply:
This is exactly in the category of "we currently expect to fail, but we will (or at least *should*) fix that".  Normally catching an exception means to me that the exception is the desired behavior, but here it represents an incomplete implementation.